### PR TITLE
Complex comparison triggers

### PR DESCRIPTION
**Pull request by my4ng**
This PR adds support for special comparison triggers, i.e. "comparison trigger that needs a parsable string parameter that will return its value in the context it is used", in the `"trigger(arg)"` syntax. 

For example, `"cultural_acceptance(scope:target_culture)" > 50`.

### CODE REVIEW

File: src/vic3/tables/triggers.rs [hunk 1]
```diff
@@ -805,3 +810,24 @@ const TRIGGER: &[(Scopes, &str, Trigger)] = &[
...
+        }
+        hash
+    });
+
+/// LAST UPDATED VIC3 VERSION N/A
+/// See `triggers.log` from the game data dumps
+/// `(inscopes, trigger name, argtype)`
+/// Currently only works with single argument triggers
+// TODO Verify triggers
+const TRIGGER_SPECIAL_VALUE: &[(Scopes, &str, Trigger)] = &[];
```
Comment (amtep) [inline src/vic3/tables/triggers.rs:833]: Hmm this is a regression since you did remove them from handle_argument. Can you put those in the table?

**Review by amtep — CHANGES_REQUESTED** [review]

File: src/vic3/tables/triggers.rs (hunk 1 above)
Comment (my4ng) [reply src/vic3/tables/triggers.rs:833]: Some of the argument validation requires updating vic3 as a whole to 1.5, i.e. adding new items and scopes. I will leave out those as `UncheckedValue` for now and leave a TODO for when it is updated.

Comment (my4ng) [reply src/vic3/tables/triggers.rs]: In addition, I think entries from the `event_targets.log` which have output scope of value (such as `num_mutual_trade_route_levels_with_country` should *not* be mixed with the entries from `triggers`, but within their own table in `scopes.rs` instead.

**Comment by my4ng** [comment]
There are still a number of unresolved issues with this PR which is *not* ready for merge.

- [ ] Scope to scope with value output type (some were erroneously placed previously)
- [ ] Values stored in variables

**Review by amtep — DISMISSED** [review]

File: src/trigger.rs [hunk 2]
```diff
@@ -969,114 +974,101 @@ pub fn validate_target_ok_this(
...
-            err(ErrorKey::UnknownField).msg(msg).opt_info(opt_info).loc(part).push();
-            sc.close();
-            return;
         }
     }
     let (final_scopes, because) = sc.scopes_reason();
     if !outscopes.intersects(final_scopes | Scopes::None) {
         let part = &part_vec[part_vec.len() - 1];
         let msg = format!("`{part}` produces {final_scopes} but expected {outscopes}");
-        if part == because.token() && part.loc == because.token().loc {
```
Comment (amtep) [inline src/trigger.rs:1074]: I remember putting this in because otherwise it led to spurious extra message blocks that pointed to the same token as the original block

File: src/trigger.rs [hunk 3]
```diff
@@ -1091,65 +1083,298 @@ pub fn validate_target(token: &Token, data: &Everything, sc: &mut ScopeContext,
...
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use crate::{emit_reports, FileKind, Loc};
+
+    use super::*;
+    #[test]
+    fn test_partition() {
+        let path = "./test.txt";
```
Comment (amtep) [inline src/trigger.rs:1363]: This is better placed in tests/files/ with a descriptive name

File: src/trigger.rs [hunk 4]
```diff
@@ -1091,65 +1083,298 @@ pub fn validate_target(token: &Token, data: &Everything, sc: &mut ScopeContext,
...
+            PathBuf::from_str(path).unwrap(),
+            FileKind::Mod,
+            PathBuf::from_str(path).unwrap(),
+        );
+        loc.line += 1;
+        loc.column += 1;
+        let str = std::fs::read_to_string(path).unwrap();
+        let token = Token::new(&str, loc);
+        let partitioned = partition(&token);
+        dbg!(token);
```
Comment (amtep) [inline src/trigger.rs:1374]: Debug output is swallowed by the test framework if the test succeeds. Common practice is to use assert macros to assert that the vector and it's tokens are what's expected.

File: src/item.rs [hunk 5]
```diff
@@ -934,6 +935,8 @@ impl Item {
             #[cfg(feature = "vic3")]
             Item::DynamicCountryName => "common/dynamic_country_names/",
             #[cfg(feature = "vic3")]
+            Item::EventCategory => "events/",
```
Comment (amtep) [inline src/item.rs:938]: Since it's defined in an internal table  the path here should be "". The path should be events iff the item values are collected from data in events/

**Review by amtep — CHANGES_REQUESTED** [review]
Looks good :) Just some small notes. I also haven't tried it yet.

Housekeeping note: I prefer to resolve conflicts by rebasing the feature branch, not by making changes in a merge commit.

File: src/trigger.rs (hunk 4 above)
Comment (my4ng) [reply src/trigger.rs:1374]: I am more or less using this as a test playground to see if the output (including the error reports) meet my expectation, so kind of difficult to use `assert` here

File: src/trigger.rs (hunk 3 above)
Comment (my4ng) [reply src/trigger.rs:1364]: It's going to be difficult as a lot of the functions and struct types are private

Comment (amtep) [reply src/trigger.rs]: I meant the file not the code :)

File: ck3-tiger/src/bin/scan-mod-ck3-tiger.rs [hunk 6]
```diff
@@ -47,7 +47,7 @@ fn main() -> Result<()> {
     everything.load_all();
 
     let mut grand_json = json!({});
-    for itype in Item::iter() {
+    for itype in <Item as IntoEnumIterator>::iter() {
```
Comment (amtep) [inline ck3-tiger/src/bin/scan-mod-ck3-tiger.rs:50]: What does this do?

File: src/validate.rs [hunk 7]
```diff
@@ -15,7 +15,7 @@ use crate::game::Game;
 use crate::item::Item;
 use crate::lowercase::Lowercase;
 use crate::report::{err, error, fatal, old_warn, report, warn, Confidence, ErrorKey, Severity};
-use crate::scopes::{scope_prefix, scope_to_scope, Scopes};
+use crate::scopes::{scope_prefix, scope_to_scope};
```
Comment (amtep) [inline src/validate.rs:18]: Why this change? I prefer to import everything at the top so that there are no surprises

Comment (my4ng) [reply src/validate.rs:18]: It is unused import when building vic3-tiger

Comment (amtep) [reply src/validate.rs]: I see :)

**Review by amtep — APPROVED** [review]
